chain/watcher.go

```diff
@@ -108,27 +110,12 @@ func (c *Watcher) maybeIndexTipSet(ctx context.Context, ts *types.TipSet) error
 	select {
 	case <-ctx.Done():
 		return ctx.Err()
-	case c.indexSlot <- struct{}{}:
-		// Indexing slot was available which means we can continue.
-		go func() {
-			// Clear the slot when we have completed indexing
-			defer func() {
-				<-c.indexSlot
-			}()
-
+	default:
+		c.wp.Submit(func() {
 			if err := c.obs.TipSet(ctx, ts); err != nil {
```

Comment:
The window value passed to the indexer will cause this method to exit when the timeout occurs.